utils/urlParams/index.js

```diff
@@ -0,0 +1,32 @@
+import { PAGE_HOME, PAGE_SPEED_UP } from '../../constants'
+
+export const persistKeyUrlParams = router => {
+  const keyParams = new Set([
+    'network'
+  ]);
+
+  const newParams = {}
+
+  const currParams = new URLSearchParams(router.query)
+  currParams.forEach((val, key) => {
+    if(keyParams.has(key)) {
+      newParams[key] = val
+    }
+  })
+
+  return new URLSearchParams(newParams)
+}
+
+export const gotoRouteWithKeyUrlParams = (router, routeUrl, extraParams) => {
+  const newParams = persistKeyUrlParams(router);
```

Comment:
Same thing here - why not use native URLSearchParams methods?

Reply:
newParams.append is a native URLSearchParams function actually. The rest of the logic is abstracting a solution for how we are always copying urlParams. Before we were always copying all the params. But we need to start only coping key params that should be copied -- just 'network' right now. 

If you knew that we'd always care only about network, then we could simplify things just a little bit, but not much.

Or maybe im missing something here... Keep in mind that this could let us replace more cases of code that has these two lines:
      const params = new URLSearchParams(router.query)
      router.push(`/home?${params.toString()}`)
      
With this helper function that gives more careful control of how we copy parameters.

Reply:
In other words, zooming out, 
as we add more routing with transaction ids or other things in the url params, I think we'll need to fix more buttons that right now are assuming we just copy all the params all the time.